AirCasting/SessionViews/MeasurementsView.swift

```diff
@@ -0,0 +1,104 @@
+// Created by Lunar on 07/06/2021.
+//
+
+import SwiftUI
+
+struct ABMeasurementsView: View {
+    
+    @ObservedObject var session: SessionEntity
+    var threshold: SensorThreshold
+    @Binding var selectedStream: MeasurementStreamEntity?
+    
+    private var streamsToShow: [MeasurementStreamEntity] {
+        let allStreams = [session.pm1Stream,
+                          session.pm2Stream,
+                          session.pm10Stream,
+                          session.FStream,
+                          session.HStream]
+        
+        let toShow = allStreams.compactMap { $0 }
+        return toShow
+    }
+    private var hasAnyMeasurements: Bool {
+        streamsToShow.filter { $0.latestValue != nil }.count > 0
+    }
+    
+    var body: some View {
+        if hasAnyMeasurements {
```

Comment:
please capture streamsToShow as let and move the logic from hasAnyMeasurements so the `streamsToShow` would not be called mupllitplle times.

Reply:
When doing as you suggested I have to move the body to `Group()`, and add a return. Convince me if I'm wrong but I think my initial solution looks better. 

 ```
private var streamsToShow: [MeasurementStreamEntity] {
        let allStreams = [session.pm1Stream,
                          session.pm2Stream,
                          session.pm10Stream,
                          session.FStream,
                          session.HStream]
        
        let toShow = allStreams.compactMap { $0 }
        return toShow
    }
    
    var body: some View {
        let streams = streamsToShow
        let hasAnyMeasurements = streams.filter { $0.latestValue != nil }.count > 0
        
        return Group {
            if hasAnyMeasurements {
                VStack(alignment: .leading, spacing: 5) {
                    Text("Most recefnt measurement:")
                    HStack {
                        Group {
                            ForEach(streams) { stream in
                                SingleMeasurementView(stream: stream,
                                                      value: stream.latestValue ?? 0,
                                                      threshold: threshold,
                                                      selectedStream: _selectedStream)
                            }
                        }
                        .frame(maxWidth: .infinity)
                    }
                }
            } else {
                VStack(alignment: .leading, spacing: 3) {
                    Text("Your AirBeam is gathering data.")
                        .font(Font.moderate(size: 14))
                    Text("Measurements will appear in 3 minutes.")
                        .font(Font.moderate(size: 12))
                }
                .foregroundColor(.darkBlue)
            }
        }
    }
}
```

Reply:
I'm just not sure if tradeoff is worth it. It's not like calling streamsToShow twice would make any performance issues.

Reply:
you should calculate the potential recreation every second and for every visible cell etc. But if not convinced maybe @lesniakania @pawelgil you can comment.

Reply:
Ok, I won't argue with you, I'm doing it your way. But tbh I'm not 100% convinced ;)